cobra/core/Metabolite.py

```diff
@@ -41,48 +43,58 @@ def __init__(self, id=None, formula=None,
         self._bound = 0.
 
     @property
+    def formula(self):
+        """Describes a Chemical Formula
+        A legal formula string contains only letters and numbers.
+        """
+        try:
+            return self._formula
+        except AttributeError:
+            # Handle loading old pickled classes
+            self._formula = self.__dict__['formula']
+            return self._formula
+
+    @formula.setter
+    def formula(self, formula):
+        self._formula = str(formula) if formula is not None else ''
+
+        if "*" in self._formula:
+            warn("invalid character '*' found in formula '{}'".format(
+                self._formula))
+        if "(" in self._formula or ")" in self._formula:
+            warn("invalid formula (has parenthesis) in '{}'".format(
+                self._formula))
+        for element, stoich in element_re.findall(self._formula):
+            if element not in elements_and_molecular_weights:
+                warn('{} not a valid element'.format(element))
+            try:
+                int(stoich) if stoich else 1
+            except ValueError:
+                warn('{} not a valid element count'.format(stoich))
+
+    @property
     def elements(self):
```

Comment:
This old version is way less clean, which I really don't like about it. Unfortunately, we need the old complexity to handle incorrect formulas, of which there are many :(

Can we just use the old elements property code?